ui/raidboss/data/05-shb/alliance/the_puppets_bunker.js

```diff
@@ -1,10 +1,261 @@
+      netRegex: NetRegexes.startsUsing({ id: '4FB4', capture: false }),
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Line Stack (Point Outside)',
+      },
+    },
+    {
+      // TODO: we could theoretically get combatants here when sharp turn starts casting.
+      // Sharp Turn ability ids differentiate left/right.  We could use effect ids to know group.
+      // But how would you communicate where to stand? "behind and slightly/very right"?
+      id: 'Puppet Superior Sharp Turn',
+      netRegex: NetRegexes.startsUsing({ id: '4FAB', capture: false }),
+      delaySeconds: 3,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Dodge Sword Charges',
+      },
+    },
+    {
+      id: 'Puppet Superior Precision Guided Missile',
+      netRegex: NetRegexes.startsUsing({ id: '4CF5', capture: false }),
+      suppressSeconds: 5,
+      alertText: function(data) {
+        // This gets casted by all the planes, so harder to collect.
+        // Not worth saying "on you" or meticulously telling the healer who their tank is.
+        if (data.role === 'tank' || data.role === 'healer') {
+          return {
+            en: 'Tank Buster',
+            de: 'Tank buster',
+            fr: 'Tank buster',
+            ja: 'タンクバスター',
+            cn: '坦克死刑',
+            ko: '탱버',
+          };
+        }
+        return {
+          en: 'Avoid tank cleave',
+          de: 'Tank Cleave ausweichen',
+          fr: 'Évitez le tank cleave',
+          ja: '前方範囲攻撃を避け',
+          ko: '광역 탱버 피하기',
+          cn: '远离顺劈',
+        };
+      },
+    },
+    {
+      id: 'Puppet Superior Sliding Swipe First',
+      netRegex: NetRegexes.startsUsing({ id: ['4FA[CD]', '550[DEF]', '5510'] }),
+      durationSeconds: 6,
+      preRun: function(data) {
+        data.swipe = data.swipe || [];
+        data.swipe.push({
+          '4FAC': Responses.goRight(),
+          '4FAD': Responses.goLeft(),
+          '550D': Responses.goRight(),
+          '550E': Responses.goLeft(),
+          '550F': Responses.goRight(),
+          '5510': Responses.goLeft(),
+        });
+      },
+      response: function(data, matches) {
+        if (data.swipe.length !== 1)
+          return;
+
+        // Call and clear the first swipe so we can not call it a second time below.
+        const swipe = data.swipe[0];
+        data.swipe[0] = null;
+        return swipe(data, matches);
+      },
+    },
+    {
+      id: 'Puppet Superior Sliding Swipe Others',
+      netRegex: NetRegexes.ability({ id: ['4FA[CD]', '550[DEF]', '5510'] }),
+      response: function(data, matches) {
+        if (!data.swipe)
+          return;
+
+        // The first swipe callout has been cleared to null.
+        // Deliberately skip it so that when the first swipe goes off, we call the second.
+        let swipe = data.swipe.pop();
+        if (!swipe)
+          swipe = data.swipe.pop();
+        if (!swipe)
+          return;
+        return swipe(data, matches);
```

Comment:
I was wondering how we could handle this one. Not sure what I would have come up with, but this is definitely better!